### PR TITLE
Line count based progress bar

### LINKED ISSUE #541: Would be handy to have a progressbar that uses the number of lines in the file, not the bytes
FileSizeProgressBar isn't quite right for line based inputs like hydration, timelines and the searches functionality proposed in #540 

Ideally we'd have a LinesInFileProgressBar for those cases.

### PR DESCRIPTION
**Pull request by igorbrigadir**
fix #541

### CODE REVIEW

File: twarc/decorators2.py [hunk 1]
```diff
@@ -197,6 +197,82 @@ class InvalidAuthType(Exception):
...
+        def blocks(files, size=65536):
+            while True:
+                b = files.read(size)
+                if not b:
+                    break
+                yield b
+
+        total_lines = 0
+        with open(infile.name, "r", encoding="utf-8", errors="ignore") as f:
+            total_lines = sum(bl.count("\n") for bl in blocks(f))
```
Comment (SamHames) [inline twarc/decorators2.py:238]: Suggestion 229 - 238 can be replaced with the following, as Python file IO in this case is on demand:

```python
with open(infile.name, "r", encoding="utf-8", errors="ignore") as f:
            total_lines = sum(1 for _ in f)
```

**Review by SamHames — COMMENTED** [review]
Looks good, thanks!

**Comment by igorbrigadir** [comment]
Yeah! this way is better!

**Comment by edsu** [comment]
Excited to see this!

**Comment by igorbrigadir** [comment]
I tried it on a few examples and it seemed to work for me - gonna merge anyway, and if there's problems i'll fix it before the next release!

**Pull request merged**